### PR TITLE
Testing:  Make sure all local test commands run with all npm packages up to date

### PR DESCRIPTION
**Pull request by gziolo**
@nylen had the following issue:

> ~/a8c/code/wp-calypso $ npm run test-client -- client/blocks/post-time/test/index.jsx 
> 
> > wp-calypso@0.17.0 test-client /home/james/a8c/code/wp-calypso
> > jest -c=test/client/jest.config.js "client/blocks/post-time/test/index.jsx"
> 
> Error: Jest: Failed to parse config file /home/james/a8c/code/wp-calypso/test/client/jest.config.js.
>     at parse (/home/james/a8c/code/wp-calypso/node_modules/jest-config/build/loadFromFile.js:22:15)
>     at promisify.then.data (/home/james/a8c/code/wp-calypso/node_modules/jest-config/build/loadFromF
> ile.js:26:20)

It turned out that running `npm install` fixed it. This PR adds corresponding `pre` commands to avoid such issues in the future.

It also adds a few cosmetic changes to the Jest config files.

### Testing

Execute the following commands and make sure they trigger `pre` commands and still pass:
* `npm run test-client`
* `npm run test-server`
* `npm run test-integration`

### CODE REVIEW

**Comment by nosolosw** [comment]
Tests pass and do call `pre-test`. I'm reviewing the code now.

File: test/client/jest.config.json [hunk 1]
```diff
@@ -1,9 +1,16 @@
 {
+	"moduleNameMapper": {
```
Comment (nosolosw) [inline test/client/jest.config.json]: What is this useful for?

**Comment by nosolosw** [comment]
I've reviewed the whole configs, not only your changes, and I have some questions that are potentially related to me not being familiar with jest (yet):

- Why do we use the `jest.config.js` file? Couldn't jest [load directly the json](https://facebook.github.io/jest/docs/en/configuration.html) in desktop scenarios?
- What's the difference between `modulePaths`, `rootDir`, and `roots`? Also, why are `modulePaths` different for client, server, and integration?
- Why integration has `verbose:true` but other config have it as false?
- Why do we collect coverage stats only for server tests?

File: test/client/jest.config.json (hunk 1 above)
Comment (gziolo) [reply test/client/jest.config.json]: You can provide a different implementation for a file. In this case our client config reads data from global variable exposed by the server in HTML script tag. That's why we need to provide a workaround :)

Comment (nosolosw) [reply test/client/jest.config.json]: To make I understood: does it mean that when in a client test, any call to config will be magically using this?

**Comment by gziolo** [comment]
> Why do we use the jest.config.js file? Couldn't jest load directly the json in desktop scenarios?

Last time I checked `--config` option didn't support `.json`, but IDEs use them. We should double check and remove `.js` file if `.json` works in such scenario. 

> What's the difference between modulePaths, rootDir, and roots? Also, why are modulePaths different for client, server, and integration?

`rootDir` is the same everywhere. This is just a way to use `<rootDir>` alias in other places.
`modulePaths` is another way of providing `NODE_PATH` which we use to teach `require` how to find files. It differs between server and client, and that's why we need to have split between those two :(
`roots` is a way to teach Jest where source files are located. I'm exploring that one as it seems like a way to help Jest to narrow down searching for files for code coverage and other things.

> Why integration has verbose:true but other config have it as false?

When set to true, it prints all test names on the console. It's cool when you have 100 tests, but otherwise slows everything down. We should probably pick false for everything.

> Why do we collect coverage stats only for server tests?

It doesn't make as much sense to have code coverage for integration tests, because they also cover non-production code. I might be wrong here.

In case of client code. It's on the todo list. It would be nice to enable code coverage there. I simply didn't find enough time to explore it.

**Comment by gziolo** [comment]
> My thinking was on setting verbose to false for everything if it didn't provide value just for making everything behave the same. No big deal though.

Yes, you are correct. I will update before merge 👍

**Comment by gziolo** [comment]
@nosolosw updated `verbose` flag to `false`. It turns out we can use `json` files directly. There is no need to use those `js` wrappers. Thanks for pointing it out 💯

File: test/client/jest.config.json (hunk 1 above)
Comment (gziolo) [reply test/client/jest.config.json]: Yes, this is how it works. It is not as flexible as Webpack, but provides the similar capabilities. You can use regular expressions to match files.

**Pull request merged**